JujuHelper/JujuHelper.psm1

```diff
@@ -133,6 +133,26 @@ function Invoke-FastWebRequest {
             }
         }
 
+        $fragment = $Uri.Fragment.Trim('#')
+        if ($fragment) {
+            $details = $fragment.Split("=")
+            $algorithm = $details[0]
+            $hash = $details[1]
+        }
+
+        if (!$SkipIntegrityCheck -and $fragment -and (Test-Path $OutFile)) {
+            if($algorithm -in @("SHA1", "SHA256", "SHA384", "SHA512", "MACTripleDES", "MD5", "RIPEMD160")) {
```

Comment:
I think you can remove this check and let the ValidateSet from Test-FileIntegrity throw an exception.